src/sql/workbench/contrib/query/browser/flavorStatus.ts

```diff
@@ -187,7 +187,7 @@ export class ChangeFlavorAction extends Action {
 
 	public run(): Promise<any> {
 		let activeEditor = this._editorService.activeEditorPane;
-		let currentUri = activeEditor?.input.resource?.toString();
+		let currentUri = activeEditor?.input.resource?.toString(true);
```

Comment:
This is fixing a bug where we were escaping the URI here still (we've fixed a ton of these). This mean that when the editor tried to get the provider ID from the URI it wouldn't match because values such as : were escaped. 

Other calls to getProviderIdFromUri are using connection ID (GUID) so don't have this issue